scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,218 @@
+import subprocess
+import sys
+
+ORIG_CWD = os.getcwd()  # Checkout changes cwd
+
+def test_infra(*paths):
+    """Return path relative to root of test-infra repo."""
+    return os.path.join(ORIG_CWD, os.path.dirname(__file__), '..', *paths)
+
+
+def check_output(*cmd):
+    """Log and run the command, return output, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    return subprocess.check_output(cmd)
+
+
+def check(*cmd):
+    """Log and run the command, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    subprocess.check_call(cmd)
+
+
+def sig_handler(_signo, _frame):
+    """Handle signal.SIGTERM and signal.SIGINT."""
+    print 'signo = %s, frame = %s' % (_signo, _frame)
+    check(['docker', 'stop', CONTAINER])
+
+
+def main(args):
+    """Set up env, trigger runner, handle termination. """
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-statements
+
+    # Get golang into our PATH so we can run e2e.go
+    os.environ['PATH'] += ':/usr/local/go/bin'
+
+    # dockerized-e2e-runner goodies setup
+    workspace = os.environ.get('WORKSPACE') or os.getcwd()
+    repo = os.environ.get('REPO_DIR') or os.getcwd()
+    artifacts = '%s/_artifacts' % workspace
+    os.makedirs(artifacts)
+
+    e2e_image_tag = 'v20170104-9031f1d'
+    e2e_image_tag_override = '%s/hack/jenkins/.kubekins_e2e_image_tag' % workspace
+    if os.path.isfile(e2e_image_tag_override):
+        with open(e2e_image_tag_override, 'r') as tag:
+            e2e_image_tag = tag.read()
+
+    docker_env_ignore = [
+      'GOOGLE_APPLICATION_CREDENTIALS',
+      'GOROOT',
+      'HOME',
+      'PATH',
+      'PWD',
+      'WORKSPACE'
+    ]
+
+    docker_extra_args = []
+    if (os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER') and
+            os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '/var/run/docker.sock:/var/run/docker.sock',
+          '-v', '%s:/go/src/k8s.io/kubernetes' % repo,
+          '-e', 'REPO_DIR=%s' % repo,
+          '-e', 'HOST_ARTIFACTS_DIR=%s' % artifacts
+        ])
+
+    if (os.environ.get('JENKINS_USE_LOCAL_BINARIES') and
+            os.environ.get('JENKINS_USE_LOCAL_BINARIES').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '%s/_output":/workspace/_output:ro' % workspace
+        ])
+
+    if 'KUBE_E2E_RUNNER' in os.environ:
+        docker_extra_args.extend([
+          '--entrypoint=%s', os.environ.get('KUBE_E2E_RUNNER')
+        ])
+
+    # exec
+
+    print 'Starting %s...' % CONTAINER
+
+    cmd = ['docker', 'run', '--rm',
+      '--name=%s' % CONTAINER,
+      '-v', '%s/_artifacts":/workspace/_artifacts' % workspace,
+      '-v', '/etc/localtime:/etc/localtime:ro'
+    ]
+
+    gce_ssh = '/workspace/.ssh/google_compute_engine'
+    gce_pub = '%s.pub' % gce_ssh
+    service = '/service-account.json'
+
+    cmd.extend(['-v', '%s:%s:ro' % (args.gce_ssh, gce_ssh)])
```

Comment:
Any reason not to combine all these extends into into the cmd = []  above?

I suspect that will enable you to enable the too-many-statements check